impeller/tessellator/tessellator_unittests.cc

```diff
@@ -13,99 +13,63 @@
 namespace impeller {
 namespace testing {
 
-TEST(TessellatorTest, TessellatorBuilderReturnsCorrectResultStatus) {
-  // Zero points.
-  {
-    Tessellator t;
-    auto path = PathBuilder{}.TakePath(FillType::kOdd);
-    Tessellator::Result result = t.Tessellate(
-        path, 1.0f,
-        [](const float* vertices, size_t vertices_count,
-           const uint16_t* indices, size_t indices_count) { return true; });
-
-    ASSERT_EQ(result, Tessellator::Result::kInputError);
-  }
-
-  // One point.
-  {
-    Tessellator t;
-    auto path = PathBuilder{}.LineTo({0, 0}).TakePath(FillType::kOdd);
-    Tessellator::Result result = t.Tessellate(
-        path, 1.0f,
-        [](const float* vertices, size_t vertices_count,
-           const uint16_t* indices, size_t indices_count) { return true; });
-
-    ASSERT_EQ(result, Tessellator::Result::kSuccess);
-  }
-
-  // Two points.
-  {
-    Tessellator t;
-    auto path = PathBuilder{}.AddLine({0, 0}, {0, 1}).TakePath(FillType::kOdd);
-    Tessellator::Result result = t.Tessellate(
-        path, 1.0f,
-        [](const float* vertices, size_t vertices_count,
-           const uint16_t* indices, size_t indices_count) { return true; });
-
-    ASSERT_EQ(result, Tessellator::Result::kSuccess);
-  }
-
-  // Many points.
-  {
-    Tessellator t;
-    PathBuilder builder;
-    for (int i = 0; i < 1000; i++) {
-      auto coord = i * 1.0f;
-      builder.AddLine({coord, coord}, {coord + 1, coord + 1});
-    }
-    auto path = builder.TakePath(FillType::kOdd);
-    Tessellator::Result result = t.Tessellate(
-        path, 1.0f,
-        [](const float* vertices, size_t vertices_count,
-           const uint16_t* indices, size_t indices_count) { return true; });
-
-    ASSERT_EQ(result, Tessellator::Result::kSuccess);
-  }
-
-  // Closure fails.
-  {
-    Tessellator t;
-    auto path = PathBuilder{}.AddLine({0, 0}, {0, 1}).TakePath(FillType::kOdd);
-    Tessellator::Result result = t.Tessellate(
-        path, 1.0f,
-        [](const float* vertices, size_t vertices_count,
-           const uint16_t* indices, size_t indices_count) { return false; });
-
-    ASSERT_EQ(result, Tessellator::Result::kInputError);
-  }
-}
-
 TEST(TessellatorTest, TessellateConvex) {
   {
     Tessellator t;
+    std::vector<Point> points;
+    std::vector<uint16_t> indices;
     // Sanity check simple rectangle.
-    auto pts = t.TessellateConvex(
-        PathBuilder{}.AddRect(Rect::MakeLTRB(0, 0, 10, 10)).TakePath(), 1.0);
-
-    std::vector<Point> expected = {{0, 0}, {10, 0}, {0, 10}, {10, 10}};
-    EXPECT_EQ(pts, expected);
+    t.TessellateConvexInternal(
+        PathBuilder{}.AddRect(Rect::MakeLTRB(0, 0, 10, 10)).TakePath(), points,
+        indices, 1.0);
+
+    // Note: the origin point is repeated but not referenced in the indices
+    // below
+    std::vector<Point> expected = {{0, 0}, {10, 0}, {10, 10}, {0, 10}, {0, 0}};
```

Comment:
The change here is that the first point is repeated if a contour is closed, where as it isn't repeat if it isn't closed. But we don't refernece the first point multiple times (in the index buffer) so the triangles are the same.

This is because we need to make sure that a filled shape without an explicit close includes the first point.